### PR TITLE
feat(cli): configurable format of dataset output

### LINKED ISSUE #426: cli: renku dataset list as jsonld
## Describe the bug
Dataset export as JSON-LD is broken 😢 

## Details
*Please verify and redact the details.*

**Renku version:** 0.3.4.dev40+dirty
**OS:** Darwin (Darwin Kernel Version 18.2.0: Fri Oct  5 19:41:49 PDT 2018; root:xnu-4903.221.2~2/RELEASE_X86_64)
**Python:** 3.7.0

## Additional context

```diff
--- a/renku/cli/dataset.py
+++ b/renku/cli/dataset.py
@@ -97,6 +97,12 @@ def dataset(ctx, client, datadir):
     ctx.meta['renku.datasets.datadir'] = datadir
 
     if ctx.invoked_subcommand is None:
+        from renku.models._jsonld import asjsonld
+        for dataset in client.datasets.values():
+            click.echo(asjsonld(dataset))
+
+        return
+
         table = tabulate(
             client.datasets.values(),
             headers=['short_id', 'name', 'authors_csv', 'created'],
```

- [ ] join the `Entity` and `DatasetFile`

### PR DESCRIPTION
**Pull request by jirikuncar**
# Description

Adds JSON-LD output format for datasets.

Fixes #426

### CODE REVIEW

File: renku/cli/dataset.py [hunk 1]
```diff
@@ -77,42 +77,82 @@
...
+
+    datasets = datasets or client.datasets
+    data = [
+        asjsonld(dataset, basedir=os.path.relpath('.', start=path.parent))
+        for path, dataset in datasets.items()
+    ]
+    click.echo(dumps(data, indent=2))
+
+
+FORMATS = {
```
Comment (jsam) [inline renku/cli/dataset.py]: Rename to DATASET_FORMATS ?

File: renku/models/_jsonld.py [hunk 2]
```diff
@@ -220,7 +224,8 @@ def asjsonld(
     def convert_value(v):
         """Convert special types."""
```
Comment (jsam) [inline renku/models/_jsonld.py]: What is a special type?

File: renku/models/datasets.py [hunk 3]
```diff
@@ -138,16 +122,22 @@ def _now(self):
...
-
-
 def _parse_date(value):
     """Convert date to datetime."""
     if isinstance(value, datetime.datetime):
         return value
     return parse_date(value)
 
 
+def _convert_dataset_files(value):
```
Comment (jsam) [inline renku/models/datasets.py]: Should we add absolute path here?

File: renku/cli/dataset.py [hunk 4]
```diff
@@ -77,42 +77,82 @@
...
 from click import BadParameter
 
-from renku.models._tabulate import tabulate
-from renku.models.datasets import Author
-
 from ._client import pass_local_client
 from ._echo import progressbar
 
 
+def _tabular(client, datasets=None):
```
[the remark this answers is not in the document]
Comment (jirikuncar) [reply renku/cli/dataset.py]: we can move it later to `renku/cli/_format/dataset.py` and rename `renku/cli/_format.py` to `renku/cli/_format/graph.py`

File: renku/models/_jsonld.py (hunk 2 above)
Comment (jirikuncar) [reply renku/models/_jsonld.py]: non-standard 😉

File: renku/cli/dataset.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jsam) [reply renku/cli/dataset.py]: Well lets do it right now cause I also want to merge ls-files. :)

File: renku/models/datasets.py (hunk 3 above)
Comment (jirikuncar) [reply renku/models/datasets.py]: no

Comment (jsam) [reply renku/models/datasets.py]: Why?

File: renku/models/_jsonld.py (hunk 2 above)
Comment (jsam) [reply renku/models/_jsonld.py]: Can you put a little bit less ambiguous comment? :hugs:

File: renku/cli/dataset.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jirikuncar) [reply renku/cli/dataset.py]: see #451

File: renku/cli/dataset.py (hunk 1 above)
Comment (jirikuncar) [reply renku/cli/dataset.py]: see #451

File: renku/models/datasets.py (hunk 3 above)
Comment (jirikuncar) [reply renku/models/datasets.py]: there is no need for an absolute path anywhere. we keep all paths relative to `client` or metadata file

Comment (jsam) [reply renku/models/datasets.py]: ```there is no need for an absolute path anywhere``` 

Not really true, now is it? ;)

**Pull request merged**